WcaOnRails/spec/controllers/competitions_controller_spec.rb

```diff
@@ -571,4 +571,79 @@
       end
     end
   end
+
+  describe 'GET #my_competitions' do
+    let(:delegate1) { FactoryGirl.create(:delegate) }
+    let!(:organizer1) { FactoryGirl.create(:user) }
+    let!(:organizer2) { FactoryGirl.create(:user) }
+    let!(:future_competition1) { FactoryGirl.create(:competition, :registration_open, starts: 3.week.from_now, organizers: [organizer1], delegates: [delegate1], eventSpecs: "222 333") }
+    let!(:future_competition2) { FactoryGirl.create(:competition, :registration_open, starts: 2.weeks.from_now, organizers: [organizer1], eventSpecs: "222 333") }
+    let!(:future_competition3) { FactoryGirl.create(:competition, :registration_open, starts: 1.weeks.from_now, organizers: [organizer1], eventSpecs: "222 333") }
+    let!(:past_competition1) { FactoryGirl.create(:competition, :registration_open, starts: 1.month.ago, organizers: [organizer1], eventSpecs: "222 333") }
+    let!(:past_competition2) { FactoryGirl.create(:competition, :registration_open, starts: 2.month.ago, organizers: [organizer2], delegates: [delegate1], eventSpecs: "222 333") }
+    let!(:past_competition3) { FactoryGirl.create(:competition, :registration_open, starts: 3.month.ago, organizers: [organizer2], delegates: [delegate1], eventSpecs: "222 333") }
+    let!(:user) { FactoryGirl.create :user, name: "Jan-Ove Waldner" }
+    let!(:registration1) { FactoryGirl.create(:registration, competitionId: future_competition1.id, user: user) }
+    let!(:registration2) { FactoryGirl.create(:registration, competitionId: future_competition3.id, user: user) }
+    let!(:registration3) { FactoryGirl.create(:registration, competitionId: past_competition1.id, user: user) }
+    let!(:registration4) { FactoryGirl.create(:registration, competitionId: past_competition3.id, user: organizer1) }
+    let!(:registration5) { FactoryGirl.create(:registration, competitionId: future_competition3.id, user: delegate1) }
+
+    context 'when not signed in' do
+      sign_out
+
+      it 'redirects to the sign in page' do
+        get :my_competitions
+        expect(response).to redirect_to new_user_session_path
+      end
+    end
+
+    context 'when signed in as a regular user' do
```

Comment:
This looks great, thanks for writing the test!

I know this is already pretty intense, but can you add a test for competitions that a user competed in, but didn't register for? This can happen for competitions whose registration wasn't done on the WCA website, for instance.